File: OOP_Tasks/OOP_Task6/Employee.cpp
#include "Employee.h"


Employee::Employee(int id, string name, string address, float salary)
{
	this->id = id;
	this->name = name;
	this->address = address;
	this->salary = salary;
}

Employee::Employee()
{
	id = -1;
	name = "";
	address = "";
	salary = 0;
}

/// setter and getter

int Employee::getId()
{
	return (id);
}

string Employee::getName()
{
	return (name);
}

void Employee::setName(string name)
{
	this->name = name;
}

void Employee::setAddress(string address)
{
	this->address = address;
}

string Employee::getAddress()
{
	return (address);
}

void Employee::setSalary()
{
	this->salary = salary;
}

float Employee::getSalary()
{
	return (salary);
}

void Employee::printData()
{
	cout << "Employee id: " << id <<
		"\tEmployee name: " << name <<
		"\tEmployee address: " << address <<
		"\tEmployee salary: " << salary;
}

File: OOP_Tasks/OOP_Task3/Employee.cs
abstract class Employee
{
    protected int id;
    protected string name;
    protected string address;
    protected float salary;
    
    public Employee()
    {
        id = -1;
        name = "";
        address = "";
        salary = 0;
    }
    
    public Employee(int id, string name, string address, float salary)
    {
        this.id = id;
        this.name = name;
        this.address = address;
        this.salary = salary;
    }
    
    public int Id
    {
        get { return (id);}
    }
    
    public string Name 
    {
        get { return (name);}
    }
    
    public string Address 
    {
        set { address = value; }
        get { return (address); }
    }
    
    public float Salary 
    {
        set { salary = value; }
        get { return (salary); }
    }
    
    public abstract string printData();
}